pkg/extension/workspace/example/runtime/getenvoy/runtime_test.go

```diff
@@ -17,166 +17,121 @@ package getenvoy_test
 import (
 	"bytes"
 	"fmt"
-	stdioutil "io/ioutil"
-	"os"
 	"path/filepath"
+	"testing"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/ginkgo/extensions/table"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 
-	"github.com/tetratelabs/getenvoy/pkg/common"
 	workspaces "github.com/tetratelabs/getenvoy/pkg/extension/workspace"
 	"github.com/tetratelabs/getenvoy/pkg/extension/workspace/example/runtime"
 	. "github.com/tetratelabs/getenvoy/pkg/extension/workspace/example/runtime/getenvoy"
 	"github.com/tetratelabs/getenvoy/pkg/extension/workspace/model"
-	argutil "github.com/tetratelabs/getenvoy/pkg/util/args"
+	"github.com/tetratelabs/getenvoy/pkg/test/cmd/extension"
 	ioutil "github.com/tetratelabs/getenvoy/pkg/util/io"
 )
 
-var _ = Describe("runtime", func() {
-	Describe("Run()", func() {
-
-		var backupHomeDir string
-
-		BeforeEach(func() {
-			backupHomeDir = common.HomeDir
-		})
-
-		AfterEach(func() {
-			common.HomeDir = backupHomeDir
-		})
-
-		var tempHomeDir string
-
-		BeforeEach(func() {
-			dir, err := stdioutil.TempDir("", "getenvoy-home")
-			Expect(err).NotTo(HaveOccurred())
-			tempHomeDir = dir
-		})
-
-		AfterEach(func() {
-			if tempHomeDir != "" {
-				Expect(os.RemoveAll(tempHomeDir)).To(Succeed())
-			}
-		})
-
-		BeforeEach(func() {
-			common.HomeDir = tempHomeDir
-		})
-
-		envoyPath := func() string {
-			path, err := filepath.Abs("testdata/envoy/bin/envoy")
-			Expect(err).ToNot(HaveOccurred())
-			return path
-		}
-
-		var stdout *bytes.Buffer
-		var stderr *bytes.Buffer
-
-		BeforeEach(func() {
-			stdout = new(bytes.Buffer)
-			stderr = new(bytes.Buffer)
-		})
-
-		runContext := func(workspace model.Workspace, example model.Example) *runtime.RunContext {
-			return &runtime.RunContext{
-				Opts: runtime.RunOpts{
-					Workspace: workspace,
-					Example: runtime.ExampleOpts{
-						Name:    "default",
-						Example: example,
-					},
-					Extension: runtime.ExtensionOpts{
-						WasmFile: `/path/to/extension.wasm`,
-						Config: model.File{
-							Source:  "/path/to/config",
-							Content: []byte(`{"key2":"value2"}`),
-						},
-					},
-					Envoy: runtime.EnvoyOpts{
-						Path: envoyPath(),
-					},
-				},
-				IO: ioutil.StdStreams{
-					Out: stdout,
-					Err: stderr,
+// relativeWorkspaceDir points to a usable pre-initialized workspace
+const relativeWorkspaceDir = "../configdir/testdata/workspace1"
```

Comment:
shall we have `getenvoy/testadata` rather than depending on `configdir`'s testadata?

Reply:
advantage of re-use is less things to copy/pasta on update. disadvantage is the opposite. In normal tests I often use a generic fixture so that it doesn't look like taking data from another tests, rather centralizing the fixture...

In this case I don't mind copying the data, but later we can think about this problem in general terms.